src/components/FormElements/XdrLedgerKeyPicker.tsx

```diff
@@ -0,0 +1,406 @@
+import { ReactElement, useEffect, useState } from "react";
+
+import { Button, Icon } from "@stellar/design-system";
+
+import { Select } from "@stellar/design-system";
+
+import { arrayItem } from "@/helpers/arrayItem";
+import { isEmptyObject } from "@/helpers/isEmptyObject";
+import { sanitizeObject } from "@/helpers/sanitizeObject";
+import * as StellarXdr from "@/helpers/StellarXdr";
+
+import { Box } from "@/components/layout/Box";
+import { XdrPicker } from "@/components/FormElements/XdrPicker";
+import { formComponentTemplateEndpoints } from "@/components/formComponentTemplateEndpoints";
+import { InputSideElement } from "@/components/InputSideElement";
+
+import { useIsXdrInit } from "@/hooks/useIsXdrInit";
+
+import { validate } from "@/validate";
+
+import {
+  AnyObject,
+  AssetObjectValue,
+  LedgerKeyFieldsType,
+  LedgerKeyType,
+} from "@/types/types";
+
+type MultiPickerProps = {
+  id: string;
+  onChange: (val: string[]) => void;
+  value: string[];
+  buttonLabel?: string;
+  limit?: number;
+};
+
+type XdrLedgerKeyPicker = {
+  id: string;
+  value: string;
+  onChange: (value: string) => void;
+  rightElement: ReactElement | null;
+};
+
+// https://github.com/stellar/stellar-xdr/blob/curr/Stellar-ledger-entries.x#L600
+const ledgerKeyFields: {
+  id: LedgerKeyType;
+  label: string;
+  templates: string;
+  addlFields?: AnyObject;
+}[] = [
+  {
+    id: "account",
+    label: "Account",
+    templates: "account_id",
+  },
+  {
+    id: "trustline",
+    label: "Trustline",
+    templates: "account_id,asset",
+    addlFields: {
+      asset: {
+        type: "",
+        code: "",
+        issuer: "",
+      },
+    },
+  },
+  {
+    id: "offer",
+    label: "Offer",
+    templates: "seller_id,offer_id",
+  },
+  {
+    id: "data",
+    label: "Data",
+    templates: "account_id,data_name",
+  },
+  {
+    id: "claimable_balance",
+    label: "Claimable Balance",
+    templates: "claimable_balance_id",
+  },
+  {
+    id: "liquidity_pool",
+    label: "Liquidity Pool",
+    templates: "liquidity_pool_id",
+  },
+  {
+    id: "contract_data",
+    label: "Contract Data",
+    templates: "contract,key,durability",
+  },
+  {
+    id: "contract_code",
+    label: "Contract Code",
+    templates: "transaction",
+  },
+  {
+    id: "config_setting",
+    label: "Config Setting",
+    templates: "config_setting_id",
+  },
+  {
+    id: "ttl",
+    label: "TTL",
+    templates: "key_hash",
+  },
+];
+
+export const XdrLedgerKeyPicker = ({
+  id,
+  value,
+  onChange,
+}: XdrLedgerKeyPicker) => {
+  const [selectedLedgerKey, selectLedgerKey] =
+    useState<LedgerKeyFieldsType | null>(null);
```

Comment:
Nit: To keep consistent naming convention.

```suggestion
  const [selectedLedgerKey, setSelectedLedgerKey] =
    useState<LedgerKeyFieldsType | null>(null);
```